test/components/generators/chat/test_azure.py

```diff
@@ -80,6 +81,20 @@ def test_to_dict_with_parameters(self, monkeypatch):
             },
         }
 
+    def test_pipeline_serialization_deserialization(self, tmp_path, monkeypatch):
+        monkeypatch.setenv("AZURE_OPENAI_API_KEY", "test-api-key")
+        generator = AzureOpenAIChatGenerator(azure_endpoint="some-non-existing-endpoint")
+        p = Pipeline()
+        p.add_component(instance=generator, name="generator")
+
+        with open(tmp_path / "test_pipeline.yaml", "w") as f:
+            p.dump(f)
+
+        with open(tmp_path / "test_pipeline.yaml", "r") as f:
+            q = Pipeline.load(f)
```

Comment:
I think we can avoid writing/reading from disk using `dumps`/`loads`.

Reply:
good point 👍🏽